apps/emqx/src/emqx_persistent_session_ds.erl

```diff
@@ -101,6 +99,24 @@ open_session(ClientID) ->
 add_subscription(TopicFilterBin, DSSessionID) ->
     ?WHEN_ENABLED(
         begin
+            %% N.B.: we chose to update the router before adding the subscription to the
```

Comment:
On a side note: I was thinking if it would be beneficial to have a similar concept of [GHC's Notes](https://gitlab.haskell.org/ghc/ghc/-/wikis/commentary/coding-style#2-using-notes) to document some design decisions in the code itself (via comments/module docs) and reference those with identifiers.  :thinking:

Reply:
I think it would be useful